### ISSUE TITLE
Zig issue when building

### ISSUE THREAD

**Issue by andreapesci** [ifreund/river#413]
Hello!

I just set up a fresh Ubuntu 21.10 machine. When i try to install river with this command:

`zig build -Drelease-safe --prefix ⁓/.local install`

i get this error:

`/home/ubuntu/river/deps/zig-wayland/src/scanner.zig:995:13: error: unused function parameter
			options: std.fmt.FormatOptions,`

i have all the prerequisites installed, and the zig version is 0.9.0-dev.866+6a6e2cd64, installed through SNAP.

**Comment by erikrodrgz** [comment]
> > Just to document, I'm reproducing the same error with zig 0.8.1-1.
> 
> I'm not sure what you call 0.8.1-1 but latest stable version is 0.8.1 and AFAIK `unused function parameter error` was introduce in 0.9.0+ so it's not possible to have this with 0.8.1

Interesting, I am using $zig version = 0.8.1, from fedora.

**Comment by Leon-Plickat** [comment]
First make sure you use rivers master branch and that all your dependencies are up to date.

```sh
git fetch --all
git reset --hard origin/master
git submodule update --init
```

Also make sure you don't accidentally have a newer zig version in your `$PATH` somewhere.

River builds fine for me with zig 0.8.1, I just tested it again.

**Comment by ifreund** [comment]
> > > Just to document, I'm reproducing the same error with zig 0.8.1-1.
> > 
> > 
> > I'm not sure what you call 0.8.1-1 but latest stable version is 0.8.1 and AFAIK `unused function parameter error` was introduce in 0.9.0+ so it's not possible to have this with 0.8.1
> 
> Interesting, I am using $zig version = 0.8.1, from fedora.

*sigh*, fedora is packaging a zig master version but calling it `0.8.1` since fedora 35. See the llvm13 patch here: <https://src.fedoraproject.org/rpms/zig/tree/rawhide>

**Comment by ifreund** [comment]
Correction: fedora isn't packaging master zig, they backported zig master commits updating to llvm13 but accidentally backported some breaking changes as well. I've contacted the maintainer of the fedora package for zig and they are working on a fix. 

In the meantime, I recommend using a static zig build from https://ziglang.org/download

**Comment by Jan200101** [comment]
Fedora package maintainer here, changes have accidentaly landed in Fedora 35 (and above) that should not have landed there.

The issue has been fixed and a new build for Fedora Rawhide is going out today but the Fedora 35 build is going to be delayed because of LLVM 13 rc1 bugs as well as the current Final Freeze until November 2nd.

**Comment by Jan200101** [comment]
`zig-0.8.1-4.fc35` has been released on Fedora which fixes the problems
On rawhide the issue has been fixed with the `zig-0.8.1-3.fc36` release
